src/compile/layer.ts

```diff
@@ -42,8 +42,13 @@ export class LayerModel extends Model {
 
     super(spec, parent, parentGivenName, config);
 
-    this.width = spec.width;
-    this.height = spec.height;
+    this.width = spec.width !== undefined ? spec.width :
```

Comment:
I don't think this is sufficient. What about layer of layer of layer? I think you need to walk up the tree until you find a size definition similar to what @arvind did for selections. Can we use the same logic?